### ISSUE TITLE
upload blocked by bad package linting

### ISSUE THREAD

**Issue by bos** [haskell/cabal#2586]
I just tried to upload a new `text` release:

```
$ cabal upload dist/text-1.1.1.4.tar.gz
Uploading dist/text-1.1.1.4.tar.gz...
Error: dist/text-1.1.1.4.tar.gz: 400 Bad Request
Error: Invalid package

'ghc-options: -Wall -Werror' makes the package very easy to break with future
GHC versions because new GHC versions often add new warnings. Use just
'ghc-options: -Wall' instead.
```

Whatever checks the package for flags isn't handling conditionals correctly: the use of `-Werror` is gated on the `developer` flag, which defaults to `False`.

Hackage used to not notice `-Werror` at all, so catching it incorrectly is a regression. There also seems to be no way to bypass well-intentioned but incorrect blocking checks like this, which is itself bad.

Since I use the conditional-`Werror` pattern basically everywhere, this blocks *all* uploads of *all* of my packages :-(

**Comment by bergmark** [comment]
https://github.com/haskell/cabal/issues/2527

**Comment by bos** [comment]
Duplicate of #2527.
